### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Movgto**
- Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.
- Used Capybara to write integration tests for each view in the project.

### CODE REVIEW

File: spec/system/users_views_spec.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    it 'redirects to the user page when clicking the link of an user' do
+      visit users_path
+      sleep(2)
+      user_link = find(:css, '.user-nav-link')
+      user_link.click do
+        expect(page).to have_current_path("/users/#{user.id}/posts")
+      end
+    end
+  end
```
Comment (codepantha) [inline spec/system/users_views_spec.rb]: - [ ] The test requires that you should assert that the user is redirected to the "user show page" not the "user post index page" as you have done here. Please ensure that there is a redirect to the appropriate page 👍

File: spec/system/users_views_spec.rb [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  context 'show' do
+    let(:user) { User.new(name: 'Diego', bio: 'Hi I\'m a student from Mexico', photo: 'https://as1.ftcdn.net/v2/jpg/00/64/30/06/1000_F_64300613_1TYCvyzq09Go7oEVXkDQypofuaJEwKTo.jpg') }
+    let(:post1) { Post.new(author: user, title: 'Post 1', text: 'Post content') }
+    let(:post2) { Post.new(author: user, title: 'Post 2', text: 'Post content') }
+    let(:post3) { Post.new(author: user, title: 'Post 3', text: 'Post content') }
+    let(:comment1) { Comment.new(user:, post: post1, text: 'Comment 1') }
+    let(:comment2) { Comment.new(user:, post: post2, text: 'Comment 2') }
+    let(:comment3) { Comment.new(user:, post: post3, text: 'Comment 3') }
```
Comment (codepantha) [inline spec/system/users_views_spec.rb]: - [ ] According to requirements, you should have separate spec files for both `User index page` and `User show page`. Please ensure that the tests covering these separate pages are in their separate files 👍

File: spec/system/users_views_spec.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    end
+
+    it 'clicking the button to see all posts redirects the user to the posts index page' do
+      visit "/users/#{user.id}"
+
+      click_link 'See all posts' do
+        expect(page).to have_current_path("/users/#{user.id}/posts")
+      end
+    end
+  end
```
Comment (codepantha) [inline spec/system/users_views_spec.rb]: - [ ] Please add a test that redirects to the post's show page when you click on a user's post 👍 
image

File: spec/system/posts_views_spec.rb [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      visit path
+
+      expect(page).to have_content(post1.title)
+    end
+
+    it 'can see the user profile picture' do
+      visit path
+
+      expect(page).to have_content(user.name)
+    end
```
Comment (codepantha) [inline spec/system/posts_views_spec.rb]: - [ ] The test description here does not match with the assertion. I think you meant `it 'can see who wrote the post' do`

File: spec/system/posts_views_spec.rb [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+      visit path
+
+      expect(page).to have_content("Likes: #{post1.likes_counter}")
+    end
+
+    it 'can see how many likes it has' do
+      visit path
+
+      expect(page).to have_content(post1.text)
+    end
```
Comment (codepantha) [inline spec/system/posts_views_spec.rb]: - [ ] I think you meant to assert that "you can see the post body" here. Please change the description of this test 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi team 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors 🟢 

## Required Changes ♻️
_Check the comments under the review._

- [ ] Please fix n+1 problems and include the before and after screenshots of your app's logs in your PR description 👍 

image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @Movgto, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**